app/model/entity/base/FlagAccessor.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace App\Model\Entity;
+
+use Exception;
+use ReflectionClass;
+
+trait FlagAccessor {
```

Comment:
Hmm... nice trait! :)

Reply:
Well it felt a bit off to just copy `getFlag` and `setFlag` methods from `UserSettings` to `AssignmentSolution` :-)